### PR TITLE
fix: Upgrade the golang version and indirect dep net, grpc

### PR DESCRIPTION
**Pull request by harshad16**
fix: Upgrade the golang version and indirect dep net, grpc

## Description
Updated the following modules:
- `golang.org/x/net v0.17.0`
- ` google.golang.org/grpc v1.56.3`

Related-to: #217 

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: components/notebook-controller/go.mod [hunk 1]
```diff
@@ -43,27 +44,27 @@ require (
 	github.com/modern-go/reflect2 v1.0.2 // indirect
 	github.com/nxadm/tail v1.4.8 // indirect
 	github.com/pkg/errors v0.9.1 // indirect
-	github.com/prometheus/client_model v0.2.0 // indirect
+	github.com/prometheus/client_model v0.3.0 // indirect
 	github.com/prometheus/common v0.28.0 // indirect
 	github.com/prometheus/procfs v0.6.0 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
 	go.uber.org/atomic v1.7.0 // indirect
 	go.uber.org/multierr v1.6.0 // indirect
```
Comment (shalberd) [inline components/notebook-controller/go.mod:52]: we could upgrade multierr to v1.10.0, which is fine with go 1.19

**Comment by harshad16** [comment]
Great question, 
so the issue with unit test was from setup-envtest as we found out.
the deeper dive explains, that it is using module 
https://github.com/kubernetes-sigs/controller-runtime/blob/cc2a25b511876693d61c8ad07936824647a8f233/tools/setup-envtest/go.mod#L20

which based on the release note has dropped the atomic bool.
https://github.com/uber-go/multierr/releases/tag/v1.10.0

to get this fixed to latest version, we need to find and fix the package which uses that atomic bool 
https://github.com/opendatahub-io/kubeflow/issues/203#issue-1946491289
and then only we can upgrade.
as that is not pressing for us. 
i would say we would look at that in near future. Hope this explains.

**Comment by shalberd** [comment]
ah, yes, I completely forgot about atomic bool, point taken.

Yeah, let's leave it as is then. I am also shooting for go [v1.19, by the way, in my PR 133](https://github.com/opendatahub-io/kubeflow/pull/133/files#diff-61128f5dd15ffdd3807b37ac7adb11cd31ae10eef290289823d47ca458f714af)

**Comment by harshad16** [comment]
Test the notebook-controller deployment with:
- quay.io/opendatahub/kubeflow-notebook-controller:pr-218
- quay.io/opendatahub/odh-notebook-controller:pr-218

Seems to be working :100: 
Merging this PR shortly.

**Comment by shalberd** [comment]
sweet, nice work on getting vulnerabilties out of the system with regards to http/2 and gRPC and the network stack in general

**Comment by shalberd** [comment]
Same here, working for me, too.

**Comment by atheo89** [comment]
It works for me as well!

**Pull request merged**